src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,21 @@
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidUserException("User must not be null");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidUserException("A user with this login is already registered");
+        }
+        if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new InvalidUserException("Login must contain at least 6 characters");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new InvalidUserException("Password must contain at least 6 characters");
+        }
+        if (user.getAge() == null || user.getAge() < 18) {
+            throw new InvalidUserException("You are not have 18 years old");
```

Comment:
Avoid hardcoding the minimum age value directly in the error message. Move the age value to a constant field with an informative name and use it in the validation check and the error message. Refer to the checklist about moving hardcoded values to constant fields.